packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/ButtonItem/buttonItem.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import defaultClasses from './buttonItem.css';
+import { arrayOf, oneOf, string, bool } from 'prop-types';
+import { Link } from '@magento/venia-drivers';
+import resolveLinkProps from '../../resolveLinkProps';
+
+const ButtonItem = props => {
+    const {
+        buttonType,
+        link,
+        linkType,
+        openInNewTab = false,
+        text,
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft,
+        cssClasses = []
+    } = props;
+
+    const dynamicInnerStyles = {
+        textAlign,
+        border,
+        borderColor,
+        borderWidth,
+        borderRadius,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft
+    };
+
+    const cssButtonTypeSuffix =
+        buttonType.charAt(0).toUpperCase() + buttonType.substring(1);
+
+    let linkProps = {};
+    let LinkComponent = 'div';
+    if (typeof link === 'string') {
+        linkProps = resolveLinkProps(link, linkType);
+        LinkComponent = linkProps.to ? Link : 'a';
+    }
+
+    return (
+        <div className={cssClasses.join(' ')}>
+            <LinkComponent
+                {...linkProps}
+                className={defaultClasses['button' + cssButtonTypeSuffix]}
+                {...(openInNewTab ? { target: '_blank' } : '')}
```

Comment:
`target="_blank"` links [should have `rel="noopener"`](https://developers.google.com/web/tools/lighthouse/audits/noopener) to prevent external pages from getting access to a pwa-studio page's `window`

Reply:
This will be resolved as part of the "Buttons" story: https://jira.corp.magento.com/browse/PB-12

Reply:
wfm 👍